### PR TITLE
feat($compile): support omitting required controller name if same as the local name

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
You have to explicitly specify the required controller's name, even if it's the same with the local name used to bind to the requiring controller (unlike how `scope: {...}` works, for example).

**What is the new behavior (if this is a feature change)?**
`require: {someDir: '?^someDir'}` can be shortened as `require: {someDir: '?^'}`.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: test/ng/compileSpec.js [hunk 1]
```diff
@@ -6268,6 +6268,51 @@ describe('$compile', function() {
...
+        siblingController = this.sibling;
+      };
+      spyOn(MeController.prototype, '$onInit').and.callThrough();
+
+      angular.module('my', [])
+        .directive('me', function() {
+          return {
+            restrict: 'E',
+            scope: {},
+            require: { parent: '^', sibling: '' },
```
Comment (petebacondarwin) [inline test/ng/compileSpec.js]: Perhaps worth checking "optional" require too?

Comment (petebacondarwin) [inline test/ng/compileSpec.js]: And `^^`

Comment (gkalpak) [inline test/ng/compileSpec.js]: I added more checks :wink:

**Comment by petebacondarwin** [comment]
LGTM

**Pull request closed without merging**